app_dart/test/service/scheduler_test.dart

```diff
@@ -450,18 +451,105 @@ void main() {
         verify(mockGithubChecksUtil.createCheckRun(any, any, any, any)).called(1);
       });
 
-      test('rerequested triggers a single luci build', () async {
-        when(mockGithubChecksUtil.createCheckRun(any, any, any, any)).thenAnswer((_) async {
-          return CheckRun.fromJson(const <String, dynamic>{
-            'id': 1,
-            'started_at': '2020-05-10T02:49:31Z',
-            'check_suite': <String, dynamic>{'id': 2},
-          });
+      test('rerequested presubmit check triggers presubmit build', () async {
+        // Note that we're not inserting any commits into the db, because
+        // only postsubmit commits are stored in the datastore.
+        config = FakeConfig(dbValue: db);
+        db = FakeDatastoreDB();
+
+        // Set up mock buildbucket to validate which bucket is requested.
+        final MockBuildBucketClient mockBuildbucket = MockBuildBucketClient();
+        when(mockBuildbucket.batch(any)).thenAnswer((i) async {
+          return FakeBuildBucketClient().batch(i.positionalArguments[0]);
```

Comment:
If you're using the fake under the hood on the mock, why not remove the mock?

Reply:
So, without that, the test is failing with an error saying `.batch` is not stubbed. And I'm using `MockBuildBucketClient` because it lets me set an expectation a few lines down for whether the `try` bucket or `prod` bucket was used, within a mocked `scheduleBuild`. I'm not sure about how I could test this with a fake.